Sources/MessageIndex.php

```diff
@@ -88,8 +88,35 @@ function MessageIndex()
 		$context['unapproved_posts_message'] = sprintf($txt['there_are_unapproved_topics'], $untopics, $unposts, $scripturl . '?action=moderate;area=postmod;sa=' . ($board_info['unapproved_topics'] ? 'topics' : 'posts') . ';brd=' . $board);
 	}
 
+	// Default sort methods.
+	$sort_methods = array(
+		'subject' => 'mf.subject',
+		'starter' => 'COALESCE(memf.real_name, mf.poster_name)',
+		'last_poster' => 'COALESCE(meml.real_name, ml.poster_name)',
+		'replies' => 't.num_replies',
+		'views' => 't.num_views',
+		'first_post' => 't.id_topic',
+		'last_post' => 't.id_last_msg'
+	);
+
+	// Default sort methods tables.
+	$sort_methods_table = array(
+		'subject' => 'JOIN {db_prefix}messages mf ON (mf.id_msg = t.id_first_msg)',
+		'starter' => 'JOIN {db_prefix}messages mf ON (mf.id_msg = t.id_first_msg)
+			LEFT JOIN {db_prefix}members AS memf ON (memf.id_member = mf.id_member)',
+		'last_poster' => 'JOIN {db_prefix}messages AS ml ON (ml.id_msg = t.id_last_msg)
+			LEFT JOIN {db_prefix}members AS meml ON (meml.id_member = ml.id_member)',
+		'replies' => '',
+		'views' => '',
+		'first_post' => '',
+		'last_post' => ''
+	);
+
+	// Bring in any changes we want to make before the query.
+	call_integration_hook('integrate_pre_messageindex', array(&$sort_methods, &$sort_methods_table));
+
 	// We only know these.
-	if (isset($_REQUEST['sort']) && !in_array($_REQUEST['sort'], array('subject', 'starter', 'last_poster', 'replies', 'views', 'first_post', 'last_post')))
+	if (isset($_REQUEST['sort']) && !in_array($_REQUEST['sort'], array_keys($sort_methods)))
```

Comment:
`if (! isset($_REQUEST['sort'])  || isset($sort_methods[$_REQUEST['sort']]))`

Reply:
Actually, the subsequent code in this function does not necessarily want `$_REQUEST['sort']` to be set at this point. It is important that if it is set at all, it is set to a valid value, but not being set is also valid.